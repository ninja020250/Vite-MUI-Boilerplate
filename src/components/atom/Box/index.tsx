import { Box as MuiBox, BoxProps as MuiBoxProps } from "@mui/material";

export type BoxProps = {
  children?: React.ReactNode;
} & MuiBoxProps;

const Box = ({ children, ...rest }: BoxProps) => {
  return <MuiBox {...rest}>{children}</MuiBox>;
};

export default Box;
