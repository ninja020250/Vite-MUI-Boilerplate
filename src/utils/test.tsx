import { render } from '@testing-library/react'
import { ThemeProvider } from '@mui/material/styles'
import theme from '@styles/theme/light'
import { ReactNode } from 'react'

const AllTheProviders = ({ children }: any) => {
  return <ThemeProvider theme={theme}>{children}</ThemeProvider>
}

const renderWithProviders = (ui: ReactNode) => {
  const tree = render(<AllTheProviders>{ui}</AllTheProviders>)
  return tree
}

export * from '@testing-library/react'

export { renderWithProviders }
