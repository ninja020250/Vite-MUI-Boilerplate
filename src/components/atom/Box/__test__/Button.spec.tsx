import { renderWithProviders } from "@utils/test";
import { it } from "vitest";
import Box from "..";

it("Render Box correctly", ({ expect }) => {
  const { asFragment } = renderWithProviders(<Box />);
  expect(asFragment()).toMatchSnapshot();
});
