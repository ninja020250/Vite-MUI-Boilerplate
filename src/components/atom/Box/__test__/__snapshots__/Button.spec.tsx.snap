// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Render Box correctly 1`] = `
<DocumentFragment>
  <div
    class="MuiBox-root css-0"
  />
</DocumentFragment>
`;
